src/platform/elements/form/DynamicForm.ts

```diff
@@ -80,129 +55,139 @@ export class NovoFieldsetElement {
+
+        if (!(this.fieldsets && this.fieldsets.length) && this.controls && this.controls.length) {
+            this.fieldsets = [{
+                controls: this.controls
+            }];
+            this.numControls = this.controls.length;
+        } else if (this.fieldsets) {
+            this.fieldsets.forEach(fieldset => {
+                this.numControls = this.numControls + fieldset.controls.length;
+            });
         }
-      });
-    });
-    this.allFieldsRequired = requiredFields.length === this.numControls;
-    this.allFieldsNotRequired = nonRequiredFields.length === this.numControls;
-    if (this.allFieldsNotRequired && this.hideNonRequiredFields) {
-      this.fieldsets.forEach(fieldset => {
-        fieldset.controls.forEach(control => {
-          this.form.controls[control.key].hidden = false;
+
+        let requiredFields: Array<any> = [];
+        let nonRequiredFields: Array<any> = [];
+        this.fieldsets.forEach(fieldset => {
+            fieldset.controls.forEach(control => {
+                if (control.required) {
+                    requiredFields.push(control);
+                } else {
+                    nonRequiredFields.push(control);
+                }
+            });
         });
-      });
-    }
-    this.form.fieldsets = [...this.fieldsets];
-  }
-
-  public showAllFields(): void {
-    this.form.fieldsets.forEach(fieldset => {
-      fieldset.controls.forEach(control => {
-        this.form.controls[control.key].hidden = false;
-      });
-    });
-    this.showingAllFields = true;
-    this.showingRequiredFields = false;
-  }
-
-  public showOnlyRequired(hideRequiredWithValue): void {
-    this.form.fieldsets.forEach(fieldset => {
-      fieldset.controls.forEach(control => {
-        // Hide any non-required fields
-        if (!control.required) {
-          this.form.controls[control.key].hidden = true;
+        this.allFieldsRequired = requiredFields.length === this.numControls;
+        this.allFieldsNotRequired = nonRequiredFields.length === this.numControls;
+        if (this.allFieldsNotRequired && this.hideNonRequiredFields) {
+            this.fieldsets.forEach(fieldset => {
+                fieldset.controls.forEach(control => {
+                    this.form.controls[control.key].hidden = false;
+                });
+            });
         }
+        this.form.fieldsets = [...this.fieldsets];
+    }
 
-        // Hide required fields that have been successfully filled out
-        if (hideRequiredWithValue && !Helpers.isBlank(this.form.value[control.key]) && (!control.isEmpty || control.isEmpty && control.isEmpty(this.form.controls[control.key]))) {
-          this.form.controls[control.key].hidden = true;
+    ngAfterContentInit() {
+        if (this.customTemplates && this.customTemplates.length) {
+            this.customTemplates.forEach((template: any) => {
+                this.templates.addCustom(template.type, template.template);
```

Comment:
Wouldn't this make defining the component on one form add it to all forms?

I would think we would want to add them independently?

Reply:
Yea, I was thinking the same. I think you should be able to do both. App level overrides/custom templates and form level overrides. What do you think?